node-lib/src/runner.rs

```diff
@@ -233,46 +235,65 @@ async fn initialize(
 }
 
 /// Processes options and potentially runs the node.
-pub async fn setup(options: Options, gui_mode: bool) -> Result<NodeSetupResult> {
+pub async fn setup(options: Options) -> Result<NodeSetupResult> {
     let command = options.command.clone().unwrap_or(Command::Mainnet(RunOptions::default()));
-    match command {
-        Command::Mainnet(run_options) => {
-            let chain_config = common::chain::config::create_mainnet();
-            start(
-                &options.config_path(*chain_config.chain_type()),
-                &options.data_dir,
-                options.create_data_dir_if_missing,
-                run_options,
-                chain_config,
-                gui_mode,
-            )
-            .await
-        }
-        Command::Testnet(run_options) => {
-            let chain_config = ChainConfigBuilder::new(ChainType::Testnet).build();
-            start(
-                &options.config_path(*chain_config.chain_type()),
-                &options.data_dir,
-                options.create_data_dir_if_missing,
-                run_options,
-                chain_config,
-                gui_mode,
-            )
-            .await
-        }
-        Command::Regtest(regtest_options) => {
-            let chain_config = regtest_chain_config(&regtest_options.chain_config)?;
-            start(
-                &options.config_path(*chain_config.chain_type()),
-                &options.data_dir,
-                options.create_data_dir_if_missing,
-                regtest_options.run_options,
-                chain_config,
-                gui_mode,
-            )
-            .await
-        }
+    let run_options = command.run_options();
+    let chain_config = match &command {
+        Command::Mainnet(_) => common::chain::config::create_mainnet(),
+        Command::Testnet(_) => common::chain::config::create_testnet(),
+        Command::Regtest(regtest_options) => regtest_chain_config(&regtest_options.chain_config)?,
+    };
+
+    // Prepare data dir
+    let data_dir = utils::default_data_dir::prepare_data_dir(
+        || default_data_dir(*chain_config.chain_type()),
+        &options.data_dir,
+        options.create_data_dir_if_missing,
+    )
+    .expect("Failed to prepare data directory");
+
+    // Lock data dir
+    let lock_file = lock_data_dir(&data_dir)?;
+
+    // Clean data dir if needed
+    if run_options.clean_data.unwrap_or(false) {
+        clean_data_dir(
+            &data_dir,
+            std::slice::from_ref(&data_dir.join(LOCK_FILE_NAME).as_path()),
+        )?;
+        return Ok(NodeSetupResult::DataDirCleanedUp);
     }
+
+    // Init logging
+    if run_options.log_to_file.is_some_and(|log_to_file| log_to_file) {
+        let log_file = FileRotate::new(
+            data_dir.join(format!("logs/{}", LOG_FILE_NAME)),
+            AppendCount::new(2),              // total 3 file
+            ContentLimit::Bytes(100_000_000), // 100MB each
```

Comment:
100 mb seems too much for one file.

IMO we should measure roughly how much logs the node produces in a day and make it the max size. And the number of files can be e.g. 7 (or 10, or 14).

Reply:
Changed to 14x10MB